python/cudf/cudf/core/column/column.py

```diff
@@ -761,13 +760,21 @@ def _check_scatter_key_length(
                 f"{num_keys}"
             )
 
+    def _scalar_to_plc_scalar(self, scalar: ScalarLike) -> plc.Scalar:
```

Comment:
This feels a bit odd as a class method. I feel like a free function that accepts a dtype would be more appropriate, then we could call that with `col.dtype`. Scoping-wise this doesn't feel like a Column method. Plus then it would directly mirror `pa_scalar_to_plc_scalar`.

Reply:
I guess there is currently a small benefit because we can override this method for decimal columns to get the specialized behavior that we need, but I think that we don't need that any more (see my comment on that class).

Reply:
Now that I've closes https://github.com/rapidsai/cudf/pull/18035 as an attempt to avoid this decimal special casing, do you still feel strongly about having this as a free function? I chose a class method because, as you mentioned, I am able to customize this for decimal and it's a little more obvious when I could remove this in the future

Reply:
No, I think it's fine to leave it as is for now.